src/main/java/com/google/graphgeckos/dashboard/datatypes/BuilderIndex.java

```diff
@@ -0,0 +1,92 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.graphgeckos.dashboard.datatypes;
+
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Entity;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Field;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.Unindexed;
+import org.springframework.data.annotation.Id;
+
+/**
+ * This class encapsulates the information needed to retain the last revision index
+ * of a builder. Since builders do not compile all revisions sequentially, they maintain
+ * an internal index of compiled revisions, which differs on a builder by builder basis.
+ * Used as a Datastore entity inside DatastoreRepository.
+ */
+@Entity(name = "index")
+public class BuilderIndex {
+  @Id
+  @Field(name = "builderName")
+  private String name;
+
+  @Unindexed
+  @Field(name = "timestamp")
+  private int index;
+
+  /**
+   * Used only by Spring GCP.
```

Comment:
Comments that say "Used only by" are dangerous because that might change over time, but readers of the documentation might start relying on this.

Reply:
Fixed it. Thank you!